src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -1051,10 +1056,13 @@ namespace WaterThermalTanks {
         int HeatingSourceNum;    // Index to DX cooling coil (heat source for desuperheater)
         int NumAlphas;           // Number of elements in the alpha array
         int NumNums;             // Number of elements in the numeric array
+        int WAHPHeatingSourceNum; // Index to WA cooling coil (heat source for desuperheater)
```

Comment:
`WAHPHeatingSourceNum` and `HeatingSourceNum` are throwaway variables that are never used. They're necessary due to the choice of functions used to check if the source type and name are a valid object, but that's all. At a minimum, these should be declared within the scope of each if block, because they have no purpose beyond that. 

Wait, these are pointless. See comments below.